modules/fs/fsw/src/cfe_fs_api.c

```diff
@@ -293,25 +293,16 @@ int32 CFE_FS_SetTimestamp(osal_id_t FileDes, CFE_TIME_SysTime_t NewTimestamp)
             CFE_FS_ByteSwapUint32(&OutTimestamp.Subseconds);
         }
 
-        Result = OS_write(FileDes, &OutTimestamp.Seconds, sizeof(OutTimestamp.Seconds));
+        Result = OS_write(FileDes, &OutTimestamp, sizeof(OutTimestamp));
 
         /* On a good write, the value returned will equal the number of bytes written */
-        if (Result == sizeof(OutTimestamp.Seconds))
+        if (Result == sizeof(OutTimestamp))
         {
-            Result = OS_write(FileDes, &OutTimestamp.Subseconds, sizeof(OutTimestamp.Subseconds));
-
-            if (Result == sizeof(OutTimestamp.Subseconds))
-            {
-                Result = OS_SUCCESS;
-            }
-            else
-            {
-                CFE_ES_WriteToSysLog("CFE_FS:SetTime-Failed to write Seconds (Status=0x%08X)\n", (unsigned int)Result);
-            }
+            Result = OS_SUCCESS;
```

Comment:
Also to be pendantic, this should be CFE_SUCCESS, and not OS_SUCCESS (assuming the status returned by this function is really a `CFE_Status_t`).

Reply:
Fixed.